### PR TITLE
Introduce WalletModel and loadWallet functionality

### PR DESCRIPTION
**Pull request by johnny9**
When a user selects a wallet from the WalletSelect menu the wallet controller can now load the wallet data in and the name and balance will appear in the WalletBadge

The WalletQmlModel is introduced that hodes the interface to the backend wallet and provides the balance to the gui as a formatted string. The formatted string isn't quite as nice as specified in the figma. Instead it just uses the satoshi formatting currently provided by our GUI utils. The advanced formatting will be added as its own PR so that it can be reviewed separately.

WalletController has been renamed to WalletQmlController to not conflict with the qt widgets controller. A function to set the selected wallet has been added. This function loads or creates the backend wallet interface and a WalletQmlModel owns the interface and is set to the controller's selectedWallet property. This is how the gui will gain access to the wallet's information. 

Loading encrypted wallets is not currently handled as we need additional dialogs. This will be done in a separate PR so that it can be reviewed independantly.

A handler is also added to the wallet controller to handle background loading of wallet either through the rpc interface or at startup.

Initial state and loading states of the wallet selector have not been implemented yet and will be done separately so it currently will just show 0 balance until a wallet is properly loaded.

### CODE REVIEW

**Comment by MarnixCroes** [comment]
forgot to push some stuff?
some things seem to be missing and CI is failing

**Review by MarnixCroes — COMMENTED** [review]
df375c0c51ae0c4b3a8b059284cedbcc685acdc1

- On regtest, I generated a wallet and did generatetoaddress 200
Then the balance was displayed like this
S

QT: 
S


- regularly it is rather slow or not working
(it's not because they are big unused wallets that need to load, it works fine when using QT)

[screencastload.webm](https://github.com/user-attachments/assets/109bc79d-ada9-4843-8bfd-89adaba94e1f)

- I've had a couple times that after closing the app the `Saving...` window stayed on for multiple minutes while there seems to be no reason as there was no heavy usage that needed to be saved, maybe related to previous point

**Review by MarnixCroes — COMMENTED** [review]
26b5ef9ae588e9b21acf1f9e05a3f498bb60049c
the negative balance issue is fixed

- Sometimes it does, but oftenly it doesn't work as supposed 

[Screencast.webm](https://github.com/user-attachments/assets/b163aa51-1815-4d4b-adff-8f0ae80c41ee)


- the balance doesn't update when balanceChanged
- the shutting down window shows all the time now (indefinitely, many minutes, have to manually kill it) when shutting down from terminal, doesn't happen when clicking the X of the app
- nit, some amount can be a bit cut off, same is for the wallet name
image

**Comment by johnny9** [comment]
update from 26b5ef9 to[ d155eed](https://github.com/bitcoin-core/gui-qml/commit/d155eedd5680cc2e07a0a932ec68cb8023ededd4)
 - Unloaded wallets when shutdown is requested.

**Comment by johnny9** [comment]
Update from d155eed to[ b41a0e4](https://github.com/bitcoin-core/gui-qml/commit/b41a0e4e05b4bc1751f207ac7673f96684764e5c)
 - Additional #ifdef to exlude functionality when --disable-wallet

**Review by MarnixCroes — COMMENTED** [review]
b41a0e4e05b4bc1751f207ac7673f96684764e5c

it sometimes happens that the walletname shown as loaded in the navigationbar is not the same as the wallet shown as loaded/selected in the wallet select dropdown

[Screencast ](https://github.com/user-attachments/assets/8b57a0ab-100e-4ce7-a080-04c04f514a0c)

as a result, clicking another wallet doesn't seem to work as it doesn't update the selected walletname in navbar

**Review by pablomartin4btc — COMMENTED** [review]
tested manually b41a0e4e05b4bc1751f207ac7673f96684764e5c in `-regtest`

As soon as `bitcoin-qt` starts, it seems that loading the latest used wallet is quite slow and takes like 5 secs to show it up on the menu.

Also it's very slow to load/ switch to another wallet (and only non/ 0 balance wallets), for wallets with some balance (using `-regtest`), doesn't work at all (selected wallet doesn't change so balance either).

Last thing for now, in Qt we have the list of wallets in the file menu to select and load and when there are more then 1 wallet loaded (multi-wallet mode) there's combo box with the loaded wallets to switch between them. I think the designers or Christoph showed me some prototype where there were small icons in the list to indicate which wallet is loaded or not, perhaps it should be another icon/ button to select only the loaded wallets, unloaded or both/ all.

**Comment by johnny9** [comment]
All of the issues commented so far are going to explode the complexity of this PR beyond anything reviewable. I just wanted to create the boilerplate for the classes that the UI will interact with. Each of the above issues should be a separate PR as thate functionality gets added to the GUI.  

The only thing this is intended to do is list the wallets in the folder and when one is selected it loads and shows the very basic formatted balance. The more advanced formatting, for instance, should be added and reviewed in its own PR as well as they asynchronous loading and event handling.

File: src/qml/models/walletqmlmodel.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+WalletQmlModel::WalletQmlModel(QObject *parent)
+    : QObject(parent)
+{
+}
+
+QString WalletQmlModel::balance() const
+{
+    if (!m_wallet) {
+        return "0";
```
[the remark this answers is not in the document]
Comment (johncarl-te) [reply src/qml/models/walletqmlmodel.cpp:25]: yeah the wallet badge needs more states and will be one of the next things to tackle in the wallet selector component.

**Pull request merged**